modules/proj4/src/lib/proj4-projection.d.ts

```diff
@@ -0,0 +1,24 @@
+import proj4 from 'proj4';
+
+/**
+ * Predefined Aliases
+ * 'WGS84':       'EPSG:4326'
+ * 'EPSG:3785':   'EPSG:3857' 'EPSG:4269'
+ * 'GOOGLE':      'EPSG:3857' 'EPSG:4269'
+ * 'EPSG:900913': 'EPSG:3857' 'EPSG:4269'
+ * 'EPSG:102113': 'EPSG:3857' 'EPSG:4269'
+ */
+export class Proj4Projection {
+  /** Define aliases for one or more projections */
+  static defineProjectionAliases(projections: {
+    [name: string]: string
+  });
+
+  /** Create a projection between two coordinate systems */
+  constructor(projections: {from?: string, to?: string});
+
+  /** Project a coordinate between two coordinate systems */
+  project(coord: number[]): number[];
```

Comment:
Not sure whether you want to allow it, but the proj4js library also allows
```ts
{x: number, y: number, z?: number, m?: number}
```

I checked and typed array input isn't allowed.

Reply:
Yes, it will generate a lot of temporary arrays. Hopefully it is not an issue, otherwise we might need to fork it to fix that.